ast-checks/src/tree_utils.rs

```diff
@@ -0,0 +1,45 @@
+use rnix::{
+    types::*, NodeOrToken, SyntaxElement, SyntaxKind, SyntaxKind::*, SyntaxNode, WalkEvent,
+};
+use std::iter::successors;
+
+pub fn walk_kind(node: &SyntaxNode, kind: SyntaxKind) -> impl Iterator<Item = SyntaxElement> {
+    node.preorder_with_tokens()
+        .filter_map(move |event| match event {
+            WalkEvent::Enter(element) => Some(element).filter(|it| it.kind() == kind),
+            WalkEvent::Leave(_) => None,
+        })
+}
+
+pub fn walk_keyvalues_filter_key(
+    node: &SyntaxNode,
+    key: &'static str,
+) -> impl Iterator<Item = KeyValue> {
+    walk_kind(node, NODE_KEY_VALUE).filter_map(move |element| match element {
+        NodeOrToken::Node(node) => {
+            let kv = KeyValue::cast(node).unwrap();
+            if kv.key().and_then(|e| Some(e.node().to_string())) == Some(key.to_string()) {
+                Some(kv)
+            } else {
+                None
+            }
+        }
+        NodeOrToken::Token(_tok) => unreachable!(""),
+    })
+}
+
+pub fn next_siblings(element: &SyntaxElement) -> impl Iterator<Item = SyntaxElement> {
+    vec![element.clone()]
+        .into_iter()
+        .chain(successors(element.next_sibling_or_token(), |it| {
+            it.next_sibling_or_token()
+        }))
```

Comment:
You're allocating twice here, can you call `.iter()` on `element` and avoid the `clone`? Maybe if you call `.cloned()` after the `.chain()`?